File: payment-assistant-bot/src/main/java/com/grabathon/paymentassistant/web/api/request/base/Payment.java
package com.grabathon.paymentassistant.web.api.request.base;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Created by Sai Ravi Teja K on 08, Dec 2019
 * © Bundl Technologies Private Ltd.
 */

public class Payment {
    @JsonProperty("id")
    public String Id;

    @JsonProperty("status")
    public String Status;

    @JsonProperty("type")
    public String Type;

    @JsonProperty("transactions")
    public Transaction[] Transactions;
}

File: wallet-service/api/request_response/payment.go
package request_response

import "grabathon/models"

/**
 * Created by Sai Ravi Teja K on 07, Dec 2019
 * © Bundl Technologies Private Ltd.
 */

type Payment struct {
	Id int
	Status string
	Type string
	Transactions []Transaction
}

func GetPaymentRequestResponse(payment models.Payment, transactions []models.Transaction) Payment {
	pmt := Payment{
		Id:           payment.Id,
		Status:       payment.Status,
		Type:         payment.Type,
		Transactions: nil,
	}

	pmt.Transactions = GetTransactionRequestResponses(transactions)

	return pmt
}

func GetPaymentModel(payment Payment) (models.Payment, []models.Transaction) {
	pmt := models.Payment{
		Id:           payment.Id,
		Status:       payment.Status,
		Type:         payment.Type,
	}

	return pmt,GetTransactionModels(payment.Transactions)
}